src/finally.js

```diff
@@ -103,9 +106,27 @@ Promise.prototype["finally"] = function (handler) {
                              finallyHandler);
 };
 
+
 Promise.prototype.tap = function (handler) {
     return this._passThrough(handler, TAP_TYPE, finallyHandler);
 };
 
+Promise.prototype.tapError = function (handlerOrPredicate) {
+    if(arguments.length === 1) {
+        return this._passThrough(handlerOrPredicate,
+                                 TAP_TYPE,
+                                 undefined,
+                                 finallyHandler);
+    } else {
+        var predicate = arguments[0];
+        var handler = arguments[1];
+       return this._passThrough(catchFilter(predicate, handler, this),
```

Comment:
It looks like `catchFilter` is expecting an array of predicates, not a single predicate; this is why the "Works with predicates" test is failing.